src/hooks/useRequestProcessor.js

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react'
+import { useHeartbeat } from './useCourtContracts'
+import { useCourtClock } from '../providers/CourtClock'
+import { useRequestQueue } from '../providers/RequestQueue'
+import { useActivity } from '../components/Activity/ActivityProvider'
+
+import radspec from '../radspec'
+import allowedTermsBehind from '../allowedTermsBehind'
+
+export function useRequestProcessor() {
+  const { request } = useHeartbeat()
+  const { addActivity } = useActivity()
+  const { addRequests } = useRequestQueue()
+  const { neededTransitions } = useCourtClock()
+
+  const processRequests = useCallback(
+    requests => {
+      const processedRequests = []
+        .concat(requests)
```

Comment:
```suggestion
      const processedRequests = requests
```

Since we're just going to map, we don't need to create an extra array and concat

Reply:
Ah actually this is necessary because the way we pass requests (can be an `Array` or`Object`) See https://github.com/aragon/court-dashboard/blob/0ec4b75a73f01a5c544c51cdbe6902f5a829d6d3/src/hooks/useCourtContracts.js#L90

Maybe we should just pass always an array?

Reply:
Ah, I missed that.

Yes, in this case, since we don't have proptypes/typescript, being a bit more obvious with the cast (either avoiding it entirely via the API, or ensuring that the reader realizes we may need to wrap into an Array) would be useful.